crates/matrix-sdk/src/sliding_sync.rs

```diff
@@ -804,9 +805,11 @@ impl SlidingSync {
                     }
                     unsubs
                 };
+                let timeout = Some(Duration::from_secs(30));
                 let req = assign!(v4::Request::new(), {
                     lists: requests,
                     pos,
+                    timeout,
```

Comment:
If you don't put the `timeout` into an `Option` when initializing, instead you put it into the `Option` on this line, then you don't need the `unwrap()` call later on for the `RequestConfig`